Makefile

```diff
@@ -0,0 +1,191 @@
+
+.PHONY : clean
+
+# compilation configuration
+$(binary_name) : export CGO_ENABLED ?= 0
+$(binary_name) : export GO111MODULE ?= on
+$(binary_name) : export GOARCH ?= amd64
+$(binary_name) : export GOOS ?= linux
+$(binary_name) : export GOPROXY ?= https://proxy.golang.org
+$(binary_name) : export revision ?= $(git_current_revision)
+
+# compile binary
+$(binary_name) : main.go exec.go go.sum
+	@$(go) build -a -ldflags "-X main.rev=$(revision)"
+
+# docker build configuration
+docker-image : export docker_build_context ?= $(docker_default_build_context)
+docker-image : export docker_file ?= $(docker_default_file)
+docker-image : export docker_repo_name ?= $(docker_default_repo_name)
+docker-image : export docker_tag ?= $(docker_default_tag)
+docker-image : export gcloud_sdk_version ?= $(gcloud_default_sdk_version)
+
+# build docker image
+docker-image : $(binary_name) $(docker_file)
+	@$(docker) build \
+		--build-arg GCLOUD_SDK_VERSION=$(gcloud_sdk_version) \
+		--tag $(docker_repo_name)/$(docker_image_name):$(docker_tag) \
+		--file $(docker_file) $(docker_build_context)
+
+.PHONY : docker-image
+
+# docker push configuration
+docker-release : export docker_repo_name ?= $(docker_default_repo_name)
+docker-release : export docker_tag ?= $(docker_default_tag)
+
+# push docker image
+docker-release :
+	@$(docker) push $(docker_repo_name)/$(docker_image_name):$(docker_tag)
+
+.PHONY : docker-release
+
+# compile binary, build and push docker image
+dist : docker-image docker-release
+
+# local docker run configuration
+docker-run-local : export CONFIG_HOME ?= $(CURDIR)/local-example
+docker-run-local : export GOOGLE_APPLICATION_CREDENTIALS ?= xxx
+docker-run-local : export PLUGIN_CLUSTER ?= yyy
+docker-run-local : export PLUGIN_NAMESPACE ?= drone-gke
+docker-run-local : export PLUGIN_SECRET_TEMPLATE ?= $(CONFIG_HOME)/.kube.sec.yml
+docker-run-local : export PLUGIN_TEMPLATE ?= $(CONFIG_HOME)/.kube.yml
+docker-run-local : export PLUGIN_VARS_PATH ?= $(CONFIG_HOME)/vars.json
+docker-run-local : export PLUGIN_ZONE ?= zzz
+docker-run-local : export SECRET_API_TOKEN ?= 123
+docker-run-local : export SECRET_BASE64_P12_CERT ?= "cDEyCg=="
+docker-run-local : export docker_repo_name ?= $(docker_default_repo_name)
+docker-run-local : export docker_tag ?= $(docker_default_tag)
+docker-run-local : export docker_cmd ?= --dry-run --verbose
+
+# run docker container using local-example
+docker-run-local :
+	@[ -d $(CONFIG_HOME) ] || ( echo "$(CONFIG_HOME) does not exist" ; exit 1 )
+	@[ -f $(PLUGIN_SECRET_TEMPLATE) ] || ( echo "$(PLUGIN_SECRET_TEMPLATE) does not exist" ; exit 1 )
+	@[ -f $(PLUGIN_TEMPLATE) ] || ( echo "$(PLUGIN_TEMPLATE) does not exist" ; exit 1 )
+	@[ -f $(PLUGIN_VARS_PATH) ] || ( echo "$(PLUGIN_VARS_PATH) does not exist" ; exit 1 )
+	@[ -f $(GOOGLE_APPLICATION_CREDENTIALS) ] || ( echo "$(GOOGLE_APPLICATION_CREDENTIALS) does not exist" ; exit 1 )
+	@cd $(CONFIG_HOME) ; \
+	$(docker) run \
+		--env PLUGIN_CLUSTER=$(PLUGIN_CLUSTER) \
```

Comment:
@fsouza do you have an opinion on using:

```sh
docker run --env PLUGIN_CLUSTER
```

vs

```sh
docker run --env PLUGIN_CLUSTER=${PLUGIN_CLUSTER}
```

?

my initial motivation behind explicitly setting it here was to give the user some feedback when running w/ the `-n` make flag (e.g., `make docker-run-local -n`) such that the value was printed.